src/components/AffiliateTabs/index.tsx

```diff
@@ -29,20 +32,33 @@ export class AffiliateTabs extends React.Component<any> {
     super();
     this.uiStore = rootStore.uiStore;
     this.affiliateStore = rootStore.affiliateStore;
+    this.featuresStore = rootStore.featuresStore;
     this.history = history;
   }
 
   componentDidMount() {
     window.scrollTo(0, 0);
 
-    if (this.affiliateStore.isAgreed && !this.affiliateStore.isLoaded) {
+    autorun(() => {
+      if (this.featuresStore.isLoaded && !this.featuresStore.hasAffiliate) {
+        this.history.replace(ROUTE_ROOT);
```

Comment:
not critical but, it's better to use injected router props for the history/location

Reply:
at least TODO would be nice